src/articles/articles.service.ts

```diff
@@ -11,28 +11,24 @@ export class ArticlesService {
     private articleRepository: Repository<Article>,
   ) {}
 
-  async createArticle(articleDto: ArticleDto): Promise<Article> {
-    const article = new Article();
+  async saveArticle(article: Article, articleDto: ArticleDto): Promise<Article> {
     article.title = articleDto.title;
     article.text = articleDto.text;
     return this.articleRepository.save(article);
   }
 
-  async findById(id: number): Promise<Article> {
-    const article = await this.articleRepository.findOne({id});
-    if (!article) {
-      throw new HttpException({
-        status: HttpStatus.NOT_FOUND,
-        errors: 'can\'t find article',
-      }, HttpStatus.NOT_FOUND);
-    }
-    return article;
+  async createArticle(articleDto: ArticleDto): Promise<Article> {
+    const article = new Article();
+    return this.saveArticle(article, articleDto);
   }
 
   async updateArticle(article: Article, articleDto: ArticleDto): Promise<Article> {
-    article.title = articleDto.title;
-    article.text = articleDto.text;
-    return this.articleRepository.save(article);
+    return this.saveArticle(article, articleDto);
+  }
+
+  async findById(id: number): Promise<Article> {
+    const article = await this.articleRepository.findOne({id});
+    return article ? article : null;
```

Comment:
Looks like redundant, because findOne() method will return undefined if no entity will find. Or am I mistaken?

Reply:
I'm agree with you